source/fundamentals/data-formats/codecs.txt

```diff
@@ -11,38 +11,337 @@ Codecs
    :class: singlecol
 
 
+.. _data-formats-codecs:
+
 Overview
 --------
 
-- Description 
-  - Relationship between Codec, CodecRegistry, and CodecProvider
+In this guide, you can learn about **Codecs** and the supporting classes that
+handle the encoding and decoding of Java objects to BSON data in MongoDB.
+The ``Codec`` abstraction allows you to map types and fields directly to
+your domain objects rather than to an intermediate map-based one such
+as ``Document`` or ``BsonDocument``.
+
+We describe the following interfaces and classes and provide examples of
+how you can specify custom encoding and decoding logic using them in the
+following sections:
+
+- :ref:`Codec <codecs-codec>`
+- :ref:`CodecRegistry <codecs-codecregistry>`
+- :ref:`CodecProvider <codecs-codecprovider>`
+- :ref:`Custom Codec Example <codecs-custom-example>`
+
+If you are customizing encoding and decoding logic for Plain old Java objects
+(POJOs), read our guide on :doc:`POJO Customization </fundamentals/data-formats/pojo-customization>`.
+
+.. _codecs-codec:
+
+Codec
+-----
+
+The ``Codec`` interface contains abstract methods for serializing and
```

Comment:
```suggestion
The :java-docs:`Codec <apidocs/bson/org/bson/codecs/Codec.html>` interface contains abstract methods for serializing and
```

Reply:
Should this link to the API docs for Codec interface?

Reply:
We should avoid links to API docs or other links external to the drivers/java site in the opening paragraph of a section that describes a concept. This decision is based on user and MongoDB writer feedback to keep the reader focused and instead link it later (see line 116).

Reply:
Sounds good, thanks for the response. Will try to avoid links to API docs in opening paragraphs in my writing going forward.